e2e-setup/setup.cluster-demo.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+source env.defaults
+
+# Run of Setup Steps scripts
+
+./10-infra-setup.sh
+./11-infra-mi-setup.sh
+export CLUSTER_TMPL_FILE=templates/cluster.default.json
+./20-cluster-create.arocurl.sh
+export NODEPOOL_TMPL_FILE=templates/nodepool.default.json
+export NODEPOOL_NAME=pool-one
+./30-nodepool-create.arocurl.sh
```

Comment:
We need some sort of wait/poll loop here where we need to ensure the cluster is in the "ready" state before making the nodepool... hitting this error:

`HTTP/1.1 400 Bad Request
Content-Type: application/json
X-Ms-Client-Request-Id: C1475200-B927-4DD7-AEE5-DA763F5326C7
X-Ms-Error-Code: InvalidRequestContent
X-Ms-Request-Id: 987159ea-c34c-4dcf-b21a-f69ffd5eb3ce
Date: Fri, 30 May 2025 15:25:04 GMT
Content-Length: 191

{
    "error": {
        "code": "InvalidRequestContent",
        "message": "Node pools can only be created on clusters in 'ready' state, cluster requested is in 'validating' state."
    }
`